### PR TITLE
Update nuxt, @nuxt/vue-app, @nuxt/types dependencies

### LINKED ISSUE #3242: Update frontend dependencies
## Problem

Several frontend dependencies are outdated, and can be updated to a later version.

This is a meta issue. Please open one PR per item in this list:

- [ ] #3247 `nuxt`, `@nuxt/vue-app`, `@nuxt/types`:  from 2.17.0 to `^2.17.2` [changes](https://github.com/nuxt/nuxt/compare/v2.17.0...v2.17.2), [changelog for 2.17.1](https://github.com/nuxt/nuxt/releases/tag/v2.17.1), [changelog for 2.17.2](https://github.com/nuxt/nuxt/releases/tag/v2.17.2),  `vue`, `vue-server-renderer`, `vue-template-compiler`: from 2.7.14 to `^2.7.15`[changelog for 2.7.15](https://github.com/vuejs/vue/releases/tag/v2.7.15)
- [ ] #3248 `@nuxtjs/sentry`: from 7.3.0 to `^7.5.0` [changes](https://github.com/nuxt-community/sentry-module/compare/v7.3.0...v7.5.0), [changelog for 7.5.0](https://github.com/nuxt-community/sentry-module/releases/tag/v7.5.0), [changelog for 7.4.0](https://github.com/nuxt-community/sentry-module/releases/tag/v7.4.0)
- [ ] #3249 `@nuxtjs/i18n`: from 7.0.3 to `^7.3.1`[changes](https://github.com/nuxt-modules/i18n/compare/v7.0.3...v7.3.1)
- [ ] #3250 `@popperjs/core`: from `2.11.2` to `^2.11.8` [changes](https://github.com/floating-ui/floating-ui/compare/v2.11.2...v2.11.8), [2.11.8 changelog](https://github.com/floating-ui/floating-ui/releases/tag/v2.11.8)
- [ ] #3252 `clipboard`: from  2.0.8 to `^2.0.11` [changes](https://github.com/zenorocha/clipboard.js/compare/v2.0.8...v2.0.11)
- [ ] #3253 `tailwindcss`: [^3.3.3](), `@tailwindcss/typography`: [^0.5.10], `autoprefixer`: [^10.4.16](https://github.com/postcss/autoprefixer/releases/tag/10.4.16)
- [ ]  #3254 `vue-demi`: from 0.13.11 to `^0.14.6` [changes](https://github.com/vueuse/vue-demi/compare/v0.13.11...v0.14.6) - 
- [ ] #3255 `uuid`: from 8.3.2 to `^9.0.1` [changes since 8.3.2](https://github.com/uuidjs/uuid/compare/v8.3.2...v9.0.1), `@types/uuid` (^9.0.6)
- [ ] #3256 `focus-trap`: from 7.1.0 to `^7.5.4` [changes](https://github.com/focus-trap/focus-trap/compare/v7.1.0...v7.5.4)

## Description

When updating the dependency, update the version in the `frontend/package.json`, run `just install` to update the lock file, and commit both files.

For first-time contributors: please make sure to create a new branch for your PR, and not add changes to your `main` branch.

## Additional context

Many dependencies cannot be updated because we are still on Nuxt 2.

### PR DESCRIPTION
**Pull request by khushi-kothari**
## Fixes

Fixes #3242 by @obulat

## Description

Updated nuxt dependency in ``frontend/package.json`` and updated the lock file, and commited the changes.

- [X] nuxt, @nuxt/vue-app, @nuxt/types: from 2.17.0 to ^2.17.2 [changes]

Screenshot 2023-10-25 at 7 43 50 PM

[](url)

## Checklist

- [X] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [X] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [X] My commit messages follow [best practices][best_practices].
- [X] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by obulat — CHANGES_REQUESTED** [review]
Great start!

You will need to run `just node-install` at the root of the repository. This will update the lock file, [`pnpm-lock.yaml`](https://github.com/WordPress/openverse/blob/main/pnpm-lock.yaml). After you commit this file, the linting should be fixed.

**Comment by khushi-kothari** [comment]
> just node-install

After running ``just node-install``,  [pnpm-lock.yaml](https://github.com/WordPress/openverse/blob/main/pnpm-lock.yaml) was not updated. Can you point out what is the issue here? 

Screenshot 2023-10-25 at 9 34 19 PM

**Comment by obulat** [comment]
> @obulat After running `just node-install`, [pnpm-lock.yaml](https://github.com/WordPress/openverse/blob/main/pnpm-lock.yaml) was not updated. Can you point out what is the issue here?

I really don't understand why your lock file is updated, @khushi-kothari 😕 

When I ran it locally on your branch, I got the following:
```
 j node-install                                                                                                                                                   ✔  8s  18:06:56 
pnpm i
Scope: all 4 workspace projects
readPackage: typescript@^4.9.5 => typescript@^4.9.5 in dependencies of @openverse/eslint-plugin
frontend                                 |  WARN  deprecated mkdirp@0.5.1
frontend                                 |  WARN  deprecated ufo@0.7.9
frontend                                 |  WARN  deprecated async-cache@1.1.0
frontend                                 |  WARN  deprecated svgo@1.3.2
frontend                                 |  WARN  deprecated source-map-resolve@0.6.0
frontend                                 |  WARN  deprecated sourcemap-codec@1.4.8
frontend                                 |  WARN  deprecated stable@0.1.8
frontend                                 |  WARN  deprecated chokidar@2.1.8
frontend                                 |  WARN  deprecated fsevents@1.2.13
frontend                                 |  WARN  deprecated querystring@0.2.1
frontend                                 |  WARN  deprecated sane@4.1.0
frontend                                 |  WARN  deprecated urix@0.1.0
frontend                                 |  WARN  deprecated source-map-resolve@0.5.3
frontend                                 |  WARN  deprecated core-js@2.6.12
frontend                                 |  WARN  deprecated querystring@0.2.0
frontend                                 |  WARN  deprecated resolve-url@0.2.1
frontend                                 |  WARN  deprecated source-map-url@0.4.1
frontend                                 |  WARN  deprecated w3c-hr-time@1.0.2
frontend                                 |  WARN  deprecated @npmcli/move-file@1.1.2
frontend                                 |  WARN  deprecated @npmcli/fs@1.1.0
frontend                                 |  WARN  deprecated uuid@3.4.0
frontend                                 |  WARN  deprecated @babel/plugin-proposal-private-property-in-object@7.21.11
Packages: +258 -121
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++------------------------------------------------------------------------
Progress: resolved 2478, reused 2469, downloaded 0, added 1, done
node_modules/.pnpm/nuxt@2.17.2_463pqhi6eauw3mjqcyq4fuonrm/node_modules/nuxt: Running postinstall script, done in 578ms
. postinstall$ pnpm --filter ./packages/* run build
│ No projects matched the filters in ".../openverse"
└─ Done in 338ms
Done in 20.1s
just frontend/run i18n:en
pnpm run i18n:en

> @openverse/frontend@2.2.1 i18n:en .../openverse/frontend
> pnpm i18n:get-translations --en-only


> @openverse/frontend@2.2.1 i18n:get-translations .../openverse/frontend
> node src/locales/scripts/get-translations "--en-only"

Successfully saved English translation to en.json.
just frontend/run i18n:copy-test-locales
pnpm run i18n:copy-test-locales

> @openverse/frontend@2.2.1 i18n:copy-test-locales .../openverse/frontend
> cp -r test/locales/* src/locales/

.../openverse  nuxt *66 !1  git status                                                                                                                                                   ✔  22s  18:07:56 
On branch nuxt
Your branch is up to date with 'khushi-kothari/nuxt'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   pnpm-lock.yaml

no changes added to commit (use "git add" and/or "git commit -a")
```
_(I redacted the openverse file paths)_

I committed the lock file and pushed it to your branch. 

Other options to try are running `pnpm install --filter=frontend` or running `just install` from the root (`openverse/`), but I'm not sure they would work better.

**Comment by khushi-kothari** [comment]
> > @obulat After running `just node-install`, [pnpm-lock.yaml](https://github.com/WordPress/openverse/blob/main/pnpm-lock.yaml) was not updated. Can you point out what is the issue here?
> 
> I really don't understand why your lock file is updated, @khushi-kothari 😕
> 
> When I ran it locally on your branch, I got the following:
> 
> ```
>  j node-install                                                                                                                                                   ✔  8s  18:06:56 
> pnpm i
> Scope: all 4 workspace projects
> readPackage: typescript@^4.9.5 => typescript@^4.9.5 in dependencies of @openverse/eslint-plugin
> frontend                                 |  WARN  deprecated mkdirp@0.5.1
> frontend                                 |  WARN  deprecated ufo@0.7.9
> frontend                                 |  WARN  deprecated async-cache@1.1.0
> frontend                                 |  WARN  deprecated svgo@1.3.2
> frontend                                 |  WARN  deprecated source-map-resolve@0.6.0
> frontend                                 |  WARN  deprecated sourcemap-codec@1.4.8
> frontend                                 |  WARN  deprecated stable@0.1.8
> frontend                                 |  WARN  deprecated chokidar@2.1.8
> frontend                                 |  WARN  deprecated fsevents@1.2.13
> frontend                                 |  WARN  deprecated querystring@0.2.1
> frontend                                 |  WARN  deprecated sane@4.1.0
> frontend                                 |  WARN  deprecated urix@0.1.0
> frontend                                 |  WARN  deprecated source-map-resolve@0.5.3
> frontend                                 |  WARN  deprecated core-js@2.6.12
> frontend                                 |  WARN  deprecated querystring@0.2.0
> frontend                                 |  WARN  deprecated resolve-url@0.2.1
> frontend                                 |  WARN  deprecated source-map-url@0.4.1
> frontend                                 |  WARN  deprecated w3c-hr-time@1.0.2
> frontend                                 |  WARN  deprecated @npmcli/move-file@1.1.2
> frontend                                 |  WARN  deprecated @npmcli/fs@1.1.0
> frontend                                 |  WARN  deprecated uuid@3.4.0
> frontend                                 |  WARN  deprecated @babel/plugin-proposal-private-property-in-object@7.21.11
> Packages: +258 -121
> +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++------------------------------------------------------------------------
> Progress: resolved 2478, reused 2469, downloaded 0, added 1, done
> node_modules/.pnpm/nuxt@2.17.2_463pqhi6eauw3mjqcyq4fuonrm/node_modules/nuxt: Running postinstall script, done in 578ms
> . postinstall$ pnpm --filter ./packages/* run build
> │ No projects matched the filters in ".../openverse"
> └─ Done in 338ms
> Done in 20.1s
> just frontend/run i18n:en
> pnpm run i18n:en
> 
> > @openverse/frontend@2.2.1 i18n:en .../openverse/frontend
> > pnpm i18n:get-translations --en-only
> 
> 
> > @openverse/frontend@2.2.1 i18n:get-translations .../openverse/frontend
> > node src/locales/scripts/get-translations "--en-only"
> 
> Successfully saved English translation to en.json.
> just frontend/run i18n:copy-test-locales
> pnpm run i18n:copy-test-locales
> 
> > @openverse/frontend@2.2.1 i18n:copy-test-locales .../openverse/frontend
> > cp -r test/locales/* src/locales/
> 
> .../openverse  nuxt *66 !1  git status                                                                                                                                                   ✔  22s  18:07:56 
> On branch nuxt
> Your branch is up to date with 'khushi-kothari/nuxt'.
> 
> Changes not staged for commit:
>   (use "git add <file>..." to update what will be committed)
>   (use "git restore <file>..." to discard changes in working directory)
>         modified:   pnpm-lock.yaml
> 
> no changes added to commit (use "git add" and/or "git commit -a")
> ```
> 
> _(I redacted the openverse file paths)_
> 
> I committed the lock file and pushed it to your branch.
> 
> Other options to try are running `pnpm install --filter=frontend` or running `just install` from the root (`openverse/`), but I'm not sure they would work better.

Thanks for adding lock file here, I will look into what is causing this issue so next updates can be committed smoothly.

**Comment by obulat** [comment]
> Ok @obulat that is the next task and am updating vue dependencies. Why are some of the CI tests failing in the current PR?

The CI tests are failing because the `nuxt` updates have updated one of the `vue` packages, and now there is a version mismatch and Nuxt cannot be built. So, the Vue dependencies should actually be updated in this PR instead of a separate PR.

**Comment by khushi-kothari** [comment]
Updated vue dependencies. Still 2 CI tests are failing.
Lock file updated this time and all changes added in the last commit that I added.
Screenshot 2023-10-25 at 10 59 08 PM

**Comment by obulat** [comment]
Weirdly, the lock file didn't fully match the updates in `package.json`, so the CI failed (you could see the messages about frozen lock file). I ran `just install` and committed the updated lock file. Let's see if the CI passes now :)

**Comment by khushi-kothari** [comment]
Hey @obulat I have added all tasks as individual PRs. Do you need any more help from my side on this issue [#3242](https://github.com/WordPress/openverse/issues/3242)?

**Comment by obulat** [comment]
> Hey @obulat I have added all tasks as individual PRs. Do you need any more help from my side on this issue [#3242](https://github.com/WordPress/openverse/issues/3242)?

Hi, @khushi-kothari ! Thank you for creating the individual PRs. Could you please update them so that they only update the dependencies listed in the corresponding PR description? 

For instance, [Update focus-trap dependency](https://github.com/WordPress/openverse/pull/3256) should only update `focus-trap`, but if you look at the files changed, you can see that it updates all of these packages:

- "@nuxtjs/i18n": "^7.3.1",    
- "@nuxtjs/sentry": "^7.5.0",    
- "@popperjs/core": "^2.11.8",
- "@tailwindcss/typography": "^0.5.10",    
- "clipboard": "^2.0.11",    "uuid": "^9.0.1",
- "vue": "^2.7.15",    
- "@types/uuid": "^9.0.6",    
- "autoprefixer": "^10.4.16",    
- "tailwindcss": "^3.3.3",    
- "vue-demi": "^0.14.6",    
- "vue-server-renderer": "^2.7.15",
- "vue-template-compiler": "^2.7.15",

**Comment by khushi-kothari** [comment]
Hey @obulat I think as it is linting problem always there is some issue while tracking whitespace changes. I locally have the following change for some reason it is not tracked while commiting. 
Screenshot 2023-10-26 at 8 03 55 PM
local ss:
Screenshot 2023-10-26 at 7 27 50 PM

But while viewing it remotely, it is like this :
Screenshot 2023-10-26 at 7 29 06 PM

> I followed following steps in this PR :
> 1. Checked out the lock file from main to remove all unrelated changes.
> ``git checkout main -- pnpm-lock.yaml``
> 2. Ran just node-install to update the lock file.
> 3. Committed the updated lock file to the branch and pushed the changes.

Yet, linting error is thrown.

**Comment by obulat** [comment]
> Hey @obulat I think as it is linting problem always there is some issue while tracking whitespace changes. I locally have the following change for some reason it is not tracked while commiting.

I am not sure what's happening with the whitespace changes and why are they not tracked... 
The lockfile changes, on the other hand, are not simply whitespace changes. In several PRs you can see that the Vue packages are still touched in the lock files, even though you reverted the Vue changes to the `package.json` file.
I could update the lock files for you if the process I described doesn't work for you locally:

> I followed following steps in this PR :
> 
> Checked out the lock file from main to remove all unrelated changes.
> git checkout main -- pnpm-lock.yaml
> Ran just node-install to update the lock file.
> Committed the updated lock file to the branch and pushed the changes.

**Comment by khushi-kothari** [comment]
> > Hey @obulat I think as it is linting problem always there is some issue while tracking whitespace changes. I locally have the following change for some reason it is not tracked while commiting.
> 
> I am not sure what's happening with the whitespace changes and why are they not tracked... The lockfile changes, on the other hand, are not simply whitespace changes. In several PRs you can see that the Vue packages are still touched in the lock files, even though you reverted the Vue changes to the `package.json` file. I could update the lock files for you if the process I described doesn't work for you locally:
> 
> > I followed following steps in this PR :
> > Checked out the lock file from main to remove all unrelated changes.
> > git checkout main -- pnpm-lock.yaml
> > Ran just node-install to update the lock file.
> > Committed the updated lock file to the branch and pushed the changes.

Ya that will be very helpful as the process you described isn't working for me locally.

I hope my contributions were helpful. As my first experience to open sourcing, I could say I will continue doing it and will grow in the process. Thankyou for being the kindest mentor and guiding me through this issue very calmly.

**Comment by obulat** [comment]
> Ya that will be very helpful @obulat as the process you described isn't working for me locally.
> 
> I hope my contributions were helpful. As my first experience to open sourcing, I could say I will continue doing it and will grow in the process. Thankyou for being the kindest mentor and guiding me through this issue very calmly.

Thank you for your contributions, @khushi-kothari, they are very helpful! Hope you can keep contributing with Openverse ❤️ 
It is often that dependency or local set up problems are more difficult than coding :)

**Review by obulat — APPROVED** [review]

File: frontend/package.json [hunk 1]
```diff
@@ -21,6 +21,7 @@
     "prod:playwright": "pnpm i18n:copy-test-locales && pnpm prod",
     "storybook": "nuxt storybook --port=54000",
     "storybook:build": "nuxt storybook build",
+    "storybook:build-for-docs": "NODE_OPTIONS=--openssl-legacy-provider nuxt storybook build",
```
Comment (obulat) [inline frontend/package.json:24]: This script is run in GitHub CI runner (not Docker), and without `openssl-legacy-provider` it fails.

**Review by dhruvkb — APPROVED** [review]
LGTM!

**Pull request merged**